csharp/src/Ice/IPEndpoint.cs

```diff
@@ -156,27 +153,23 @@ public override void IceWritePayload(OutputStream ostr)
             ostr.WriteInt(Port);
         }
 
-        public override Endpoint NewConnectionId(string connectionId) =>
-            connectionId == ConnectionId ? this : CreateEndpoint(Host, Port, connectionId);
+        public override Endpoint NewCompressionFlag(bool compressionFlag) =>
+            compressionFlag == HasCompressionFlag ? this : CreateEndpoint(Host, Port, ConnectionId, compressionFlag,
+                Timeout);
 
-        public virtual List<IConnector> Connectors(List<IPEndPoint> endpoints, INetworkProxy? proxy)
-        {
-            var connectors = new List<IConnector>();
-            foreach (EndPoint p in endpoints)
-            {
-                connectors.Add(CreateConnector(p, proxy));
-            }
-            return connectors;
-        }
+        public override Endpoint NewConnectionId(string connectionId) =>
+            connectionId == ConnectionId ? this : CreateEndpoint(Host, Port, connectionId, HasCompressionFlag,
+                Timeout);
 
-        public override async ValueTask<IEnumerable<IConnector>> ConnectorsAsync(EndpointSelectionType endptSelection)
+        public override async ValueTask<IEnumerable<IConnector>> ConnectorsAsync(
```

Comment:
A number of these methods are Ice-internal but publicly visible and should get an Ice prefix (for another PR).
Or another option could be to convert IEndpoint into an Endpoint abstract base class and use protected internal methods.

Reply:
We need to cleanup accessibility in this API, and ensure that creating a new Transport outside is doable